Shopify/cli
# [Themes] Theme Console - Handle the REPL lifecycle

### WHY are these changes introduced?
- https://github.com/Shopify/develop-advanced-edits/issues/259

### WHAT is this pull request doing?
EDIT - _This has been modified to use `readline` to accept console inputs _
https://github.com/user-attachments/assets/0645b141-be79-4c82-aff9-84797e1fb9fb


### How to test your changes?
1) Checkout the branch
2) `pnpm shopify theme console --dev-preview`
3) Play around with the console. Note that the error messages will change as the evaluator is implemented. (There are also different layers - the shutdown message, which stops the loop (hard to trigger) and the graceful message which catches the error and keeps the loop going.

Some ideas
- Delimiter - `{{ collections.first }}` or `{% x %}`
- Null value - `f`
- Something that will error out - `JSON(`
- Please feel free to suggest more cases!


### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
